src/lib/plugins/inventory.js

```diff
@@ -1,26 +1,35 @@
+var Version = require("../version")
+var Windows = require("prismarine-windows")(Version).windows
+var ItemStack = require("prismarine-item")(Version)
+
 module.exports.player=function(player)
 {
-  player.heldItemSlot=0;
-  player.heldItem=0;
-  player.inventory=new Array(44);
+  player.heldItemSlot = 0
+  player.heldItem = new ItemStack(256, 1)
+  player.inventory = new Windows.InventoryWindow(0, "Inventory", 44)
   
   player._client.on("held_item_slot", ({slotId} = {}) => {
     player.heldItemSlot = slotId;
-    if(player.inventory[36+player.heldItemSlot]===undefined){
-      player.inventory[36+player.heldItemSlot]={
-            blockId:-1
-        };
-    }
-    player.heldItem = player.inventory[36+player.heldItemSlot];
+    
+    player.heldItem = player.inventory.itemsRange(36 + player.heldItemSlot, 36 + player.heldItemSlot + 1)
     player._writeOthersNearby("entity_equipment",{
-        entityId:player.id,
-        slot:0,
-        item:player.heldItem
+        entityId: player.id,
+        slot: 0,
+        item: ItemStack.toNotch(player.heldItem)
     });
   });
   
   player._client.on("set_creative_slot", ({slot,item} ={}) => {
-    player.inventory[slot]=item;
+    if(item.blockId == -1){
+      player.inventory.updateSlot(slot, undefined)
+      player.emit("windowUpdate", slot, player.inventory.slots[slot], undefined)
+      return;
+    }
+    
+    var NewItem = new ItemStack(item.blockId, item.itemCount, item.metadata)
+    player.emit("windowUpdate", slot, player.inventory.slots[slot], NewItem)
```

Comment:
nah, the whole point of using that event is that it is already emitted by prismarine-windows (see https://github.com/PrismarineJS/prismarine-windows/blob/master/lib/windows.js#L203)